packages/konsole-app/src/panels/RecordPanel.tsx

```diff
@@ -42,28 +41,65 @@ const useStyles = makeStyles(theme => ({
 
 const delay = 500;
 
+export interface IRecordType {
+  type: string
+  label: string
+}
+
+export function mapRecordsTypes (records: Resource[]): IRecordType[] {
+  const mapped = records.map(apiRecord => ({
+    type: apiRecord.messageFqn,
+    label: apiRecord.messageFqn
+  }));
+
+  return Object.values(Object.fromEntries(mapped.map(record => [record.type, record])));
+}
+
+export interface IRecord {
+  cid: string
+  created: number
+  name: string
+  type: string
+  title: string
+  url?: string
+}
+
+export function mapRecords (records: Resource[], config: IConfig): IRecord[] {
+  return records.map(apiRecord => ({
+    cid: apiRecord.cid.toB58String(),
+    // TODO (marcin): Currently registry API does not expose that. Add that to the DTO.
+    created: apiRecord.data?.attributes?.created ?? Date.now(),
```

Comment:
Why `Date.now()`? Can't we be honest and return `undefined`?

Reply:
sure, thx